api/loader/loader.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loader
+
+import (
+	"fmt"
+	"os"
+	"plugin"
```

Comment:
`plugin` package which is supported on Linux/MacOS only. Are we OK with such limitation?

Reply:
| "Is dynamically loading the SDK from the plugin the only option or I can still link with the SDK statically?"

I feel that we need to decide which are / are not requirements:

1) We arrange so that the global SDK object is initialized exactly once before first use.
 - It's a nice property, good for "zero touch"
 - It's a difficult property, complicates deployment and/or build and/or customization
2) We support replacing the global SDK at runtime
 - It means some applications will get a Noop SDK
 - It means an SDK could be registered and quickly unregistered
3) Require an explicit Init() call
 - Good for linking statically, good for customization
 - Bad for zero-touch instrumentation

I've found that there are utilities that support building Go binaries with resources built in, presumably we could use something like this. https://github.com/go-bindata/go-bindata

The issue for plugin support on Windows is just waiting for a contribution: https://github.com/golang/go/issues/19282

One sort of compromise would be to develop a kind of "meta" SDK that can be registered once on first use and supports dynamically attaching and detaching other SDKs. Such an SDK could be configured to buffer and replay events until the first real SDK is registered, for example. Such a "meta" SDK could be a good default implementation, supposing that generally the user will link the SDK they want and initialize it when the time is right.